packages/js/src/plugins/auctionHouseModule/operations/findBidsByPublicKeyField.ts

```diff
@@ -0,0 +1,90 @@
+import { Commitment, PublicKey } from '@solana/web3.js';
+import { UnreachableCaseError } from '@/errors';
+import { Metaplex } from '@/Metaplex';
+import { Operation, OperationHandler, useOperation } from '@/types';
+import { DisposableScope } from '@/utils';
+import { findMetadataPda } from '../../nftModule';
+import { BidReceiptGpaBuilder } from '../gpaBuilders';
+import { AuctionHouse, LazyBid, toLazyBid } from '../models';
+import { AuctionHouseProgram } from '../program';
+import { toBidReceiptAccount } from '../accounts';
+
+// -----------------
+// Operation
+// -----------------
+
+const Key = 'FindBidsByPublicKeyOperation' as const;
+
+/**
+ * @group Operations
+ * @category Constructors
+ */
+export const findBidsByPublicKeyFieldOperation =
+  useOperation<FindBidsByPublicKeyFieldOperation>(Key);
+
+/**
+ * @group Operations
+ * @category Types
+ */
+export type FindBidsByPublicKeyFieldOperation = Operation<
+  typeof Key,
+  FindBidsByPublicKeyFieldInput,
+  FindBidsByPublicKeyFieldOutput
+>;
+
+/**
+ * @group Operations
+ * @category Inputs
+ */
+export type FindBidsByPublicKeyFieldInput = {
+  type: 'buyer' | 'metadata' | 'mint';
+  auctionHouse: AuctionHouse;
+  publicKey: PublicKey;
+  commitment?: Commitment;
+};
+
+/**
+ * @group Operations
+ * @category Outputs
+ */
+export type FindBidsByPublicKeyFieldOutput = LazyBid[];
```

Comment:
Apologies for the confusion but this should still allow for `Bid` to be returned.

```suggestion
export type FindBidsByPublicKeyFieldOutput = (LazyBid | Bid)[];
```

This is because, whilst _our_ operation handlers only return `LazyBid`s other handlers could return `Bid`s directly when fetching from an indexer for example.